Orange/data/table.py

```diff
@@ -54,6 +54,21 @@ def __init__(self):
 _thread_local = _ThreadLocal()
 
 
+def _idcache_save(cachedict, keys, value):
+    # key is tuple(list) not tuple(genexpr) for speed
+    cachedict[tuple([id(k) for k in keys])] = \
+        value, [weakref.ref(k) for k in keys]
+
+
+def _idcache_restore(cachedict, keys):
+    # key is tuple(list) not tuple(genexpr) for speed
+    shared, weakrefs = cachedict.get(tuple([id(k) for k in keys]), (None, []))
```

Comment:
You can omit creating a list, `tuple(id(k) for k in keys)`, or even use `tuple(map(id, keys))`. Same in the above function.

This is, of course, an extremely imporant comment.

Reply:
Yes, at first I had `tuple(id(k) for k in keys)`, but that measured as significantly slower than `tuple(map(id, keys))`. I have no idea why though.

Reply:
I started falling into a black hole of Python source, but caught myself and stopped. This is very interesting and perhaps worth reporting if reproducable in a simple case.

Reply:
```python
tuple(map(id, keys))
```

Reply:
```import time

l = ["a", "b", "c"]

N = 1000000

t = time.time()
for i in range(N):
    tuple([id(a) for a in l])
print("tuple(list)", time.time() - t)

t = time.time()
for i in range(N):
    tuple(map(id, l))
print("tuple(map)", time.time() - t)

t = time.time()
for i in range(N):
    tuple(id(a) for a in l)
print("tuple(genexpr)", time.time() - t)
```


With Python 3.8 on Linux I get this (times can vary a bit, but usually the differences are similar).

```
tuple(list) 0.3450343608856201
tuple(map) 0.32802605628967285
tuple(genexpr) 0.4420166015625
```

So yes, I'll change to maps.